platform/lib/routers/thumbor.js

```diff
@@ -68,24 +69,40 @@ thumborRouter.get(imagePaths, async (request, response, next) => {
   thumborUrl.pathname =
     SECURITY_KEY + (imageWidth ? `/${imageWidth}x0/` : '/') + imageUrl.href;
 
+  let optimizedImage;
   try {
-    const optimizedImage = await fetch(thumborUrl.toString(), {
+    optimizedImage = await fetch(thumborUrl.toString(), {
       headers: request.headers,
     });
-    if (!optimizedImage.ok) {
-      log.error('Thumbor did not respond to', thumborUrl.toString());
-      // If Thumbor did not respond, fail over to default static middleware
-      next();
-      return;
-    }
-
-    const contentType = optimizedImage.headers.get('content-type');
-    response.setHeader('Content-Type', contentType);
-    optimizedImage.body.pipe(response);
   } catch (e) {
     log.error('Failed connecting to thumbor', e);
-    next();
+    optimizedImage = {ok: false};
   }
+  if (!optimizedImage.ok) {
+    log.error('Thumbor did not respond to', thumborUrl.toString());
+    if (isRemoteImage) {
+      // Fetch the image from remote
+      try {
+        optimizedImage = await fetch(imageUrl);
+        if (!optimizedImage.ok) {
+          log.error('Failed downloading remote image', optimizedImage.status);
+          res.status(404).send('Not found');
+          return;
+        }
+      } catch (e) {
+        log.error('Failed downloading remote image', e);
+        res.status(404).send('Not found');
+        return;
+      }
+    } else {
+      // If Thumbor did not respond, fail over to default static middleware
```

Comment:
Is that what this is doing? It looks like it would return the 404 if thumbor has a failure. Are we relying on the upstream server to determine a time out?￼ Or am I just not seeing the larger picture of how this fits in￼

Reply:
For local images we fall back to the static middleware and serve the large version of the image (this works already). What's new is that we also handle the case where we use Thumbor to optimize remote images from the blog. In that case we proxy the request.

Reply:
Ah, so in the case of an error in thumbor with remote images we send a 404, rather than 302 to the origin URL or whatever?